File: 438 Liczby Pierwsze/main.c
#include <stdio.h>
#include <stdlib.h>
#include <math.h>

int main(void)
{
	int n, i, j;
	int *tab;
	scanf("%d", &n);
	if(n > 100000)
		return 0;

	tab = malloc(sizeof(int) * n);

	for(i=0; i<n; i++)
	{
		scanf("%d", &tab[i]);
		if(tab[i] < 1 || tab[i] > 10000)
			return 0;

		for(j=2; j<=sqrt(tab[i]); j++)
		{
			if((tab[i]%j) == 0)
			{
				tab[i] = 0;
				break;
			}
		}

	}

	for(i=0; i<n; i++)
	{
		if(tab[i] == 0 || tab[i] == 1)
			puts("NIE");
		else
			puts("TAK");
	}

	free(tab);

	return 0;
}

File: 438 Liczby Pierwsze/main.js
function isPrimaryNumber(number) {
	if(number < 3) {
		return number === 2 ? "TAK" : "NIE";
	}
 
	const sqrt = Math.ceil(Math.sqrt(number));
	for(let j=2; j<=sqrt; j++) {
		if(number % j === 0) {
			return "NIE";
		}
	}
	return "TAK";
}
 
(function () {
	let n = readline();
	n = isNaN(n) || n > 100000 ? 0 : n;
	for(let i=0; i<Number(n); i++) {
		const number = readline();
		if(isNaN(number)) {
			break;
		}
		const result = isPrimaryNumber(Number(number));
		console.log(result);
	}
})()